Sonos populating opml configuration

I've had the issue a couple of times now where the Sonos binding is populating the OPML configuration field with a url and then generating the following error.  I've deleted the OPML configuration parameter a few days ago.  Uninstalled the binding, deleted the .cfg file and reinstalled and it was fine.  Just uploaded to the latest snapshot and it populated it again.

```
2017-02-10 17:21:46.324 [ERROR] [.io.transport.upnp.UpnpIOServiceImpl] - Participant threw an exception onValueReceived
java.lang.IllegalArgumentException: Malformed escape pair at index 75: http://opml.radiotime.com/Describe.ashx?c=nowplaying&partnerId=IAeIhU42&id=%id&serial=00:0E:58:AE:CB:56
	at java.net.URI.create(URI.java:852)[:1.8.0_121]
	at org.eclipse.jetty.client.HttpClient.newRequest(HttpClient.java:404)[69:org.eclipse.jetty.client:9.2.19.v20160908]
	at org.eclipse.smarthome.io.net.http.HttpUtil.executeUrl(HttpUtil.java:185)[113:org.eclipse.smarthome.io.net:0.9.0.201702100936]
	at org.eclipse.smarthome.io.net.http.HttpUtil.executeUrl(HttpUtil.java:130)[113:org.eclipse.smarthome.io.net:0.9.0.201702100936]
	at org.eclipse.smarthome.io.net.http.HttpUtil.executeUrl(HttpUtil.java:86)[113:org.eclipse.smarthome.io.net:0.9.0.201702100936]
	at org.eclipse.smarthome.io.net.http.HttpUtil.executeUrl(HttpUtil.java:65)[113:org.eclipse.smarthome.io.net:0.9.0.201702100936]
	at org.eclipse.smarthome.binding.sonos.handler.ZonePlayerHandler.updateMediaInformation(ZonePlayerHandler.java:929)[188:org.eclipse.smarthome.binding.sonos:0.9.0.201702100936]
	at org.eclipse.smarthome.binding.sonos.handler.ZonePlayerHandler.onValueReceived(ZonePlayerHandler.java:388)[188:org.eclipse.smarthome.binding.sonos:0.9.0.201702100936]
	at org.eclipse.smarthome.io.transport.upnp.UpnpIOServiceImpl$UpnpSubscriptionCallback.eventReceived(UpnpIOServiceImpl.java:144)[189:org.eclipse.smarthome.io.transport.upnp:0.9.0.201702100936]
	at org.jupnp.controlpoint.SubscriptionCallback$2.eventReceived(SubscriptionCallback.java:222)[162:org.jupnp:2.2.0]
	at org.jupnp.model.gena.RemoteGENASubscription.receive(RemoteGENASubscription.java:114)[162:org.jupnp:2.2.0]
	at org.jupnp.protocol.sync.ReceivingEvent$2.run(ReceivingEvent.java:130)[162:org.jupnp:2.2.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)[:1.8.0_121]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)[:1.8.0_121]
	at java.lang.Thread.run(Thread.java:745)[:1.8.0_121]
Caused by: java.net.URISyntaxException: Malformed escape pair at index 75: http://opml.radiotime.com/Describe.ashx?c=nowplaying&partnerId=IAeIhU42&id=%id&serial=00:0E:58:AE:CB:56
	at java.net.URI$Parser.fail(URI.java:2848)[:1.8.0_121]
	at java.net.URI$Parser.scanEscape(URI.java:2978)[:1.8.0_121]
	at java.net.URI$Parser.scan(URI.java:3001)[:1.8.0_121]
	at java.net.URI$Parser.checkChars(URI.java:3019)[:1.8.0_121]
	at java.net.URI$Parser.parseHierarchical(URI.java:3111)[:1.8.0_121]
	at java.net.URI$Parser.parse(URI.java:3053)[:1.8.0_121]
	at java.net.URI.<init>(URI.java:588)[:1.8.0_121]
	at java.net.URI.create(URI.java:850)[:1.8.0_121]
	... 14 more
```

--- a/extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/handler/ZonePlayerHandler.java
+++ b/extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/handler/ZonePlayerHandler.java
@@ -920,38 +920,41 @@ public class ZonePlayerHandler extends BaseThingHandler implements UpnpIOPartici
             boolean opmlUrlSucceeded = false;
             if (opmlUrl != null) {
                 String stationID = StringUtils.substringBetween(currentURI, ":s", "?sid");
-                String url = opmlUrl;
-                url = StringUtils.replace(url, "%id", stationID);
-                url = StringUtils.replace(url, "%serial", getMACAddress());
+                String mac = getMACAddress();
+                if (stationID != null && !stationID.isEmpty() && mac != null && !mac.isEmpty()) {
+                    String url = opmlUrl;
+                    url = StringUtils.replace(url, "%id", stationID);
+                    url = StringUtils.replace(url, "%serial", mac);
 
-                String response = null;
-                try {
-                    response = HttpUtil.executeUrl("GET", url, SOCKET_TIMEOUT);
-                } catch (IOException e) {
-                    logger.debug("Request to device failed: {}", e);
-                }
+                    String response = null;
+                    try {
+                        response = HttpUtil.executeUrl("GET", url, SOCKET_TIMEOUT);
+                    } catch (IOException e) {
+                        logger.debug("Request to device failed: {}", e);
+                    }
 
-                if (response != null) {
-                    List<String> fields = SonosXMLParser.getRadioTimeFromXML(response);
+                    if (response != null) {
+                        List<String> fields = SonosXMLParser.getRadioTimeFromXML(response);
 
-                    if (fields != null && fields.size() > 0) {
+                        if (fields != null && fields.size() > 0) {
 
-                        opmlUrlSucceeded = true;
+                            opmlUrlSucceeded = true;
 
-                        resultString = new String();
-                        // radio name should be first field
-                        title = fields.get(0);
+                            resultString = new String();
+                            // radio name should be first field
+                            title = fields.get(0);
 
-                        Iterator<String> listIterator = fields.listIterator();
-                        while (listIterator.hasNext()) {
-                            String field = listIterator.next();
-                            resultString = resultString + field;
-                            if (listIterator.hasNext()) {
-                                resultString = resultString + " - ";
+                            Iterator<String> listIterator = fields.listIterator();
+                            while (listIterator.hasNext()) {
+                                String field = listIterator.next();
+                                resultString = resultString + field;
+                                if (listIterator.hasNext()) {
+                                    resultString = resultString + " - ";
+                                }
                             }
-                        }
 
-                        needsUpdating = true;
+                            needsUpdating = true;
+                        }
                     }
                 }
             }

